streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -184,7 +184,7 @@ private KTable<K, Long> doCount(final Named named, final Materialized<K, Long, K
 
         final String name = new NamedInternal(named).orElseGenerateWithPrefix(builder, AGGREGATE_NAME);
         return doAggregate(
-            new KStreamAggregate<>(materializedInternal.storeName(), aggregateBuilder.countInitializer, aggregateBuilder.countAggregator),
+            new KStreamAggregate<>(materializedInternal, aggregateBuilder.countInitializer, aggregateBuilder.countAggregator),
```

Comment:
don't we need to make a change to either pass `null` for the store factory to `doAggregate`? Otherwise we'll wind up adding the state store to the topology again from `StatefulProcessorNode`

Reply:
I think it's okay since we also changed https://github.com/apache/kafka/pull/17929/files#diff-593acf162f2804512ad5e034f6646d6d6b61dc7f8e6904ee4e711bcb346d2fa0L94 hence calling another constructor, and this one (https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StatefulProcessorNode.java#L49) does not set store factory?

Reply:
ah sorry I forgot to respond to this earlier -- basically you're right that we ultimately need to remove it from `StateProcessorNode` but  we need to convert a bunch of other stuff to implement `#stores` before we can do this. I put `StatefulProcessorNode`  on our TODO list with a note to say it should be picked up last.

In the meantime, it doesn't hurt to add the store twice. The InternalTopologyBuilder has checks for this (see `StoreFactory#isCompatibleWith`)

Reply:
wait sorry, I was thinking of something else -- the StatefulProcessorNode does already have an alternative constructor we can use for nodes that add/connect their stores on their own. So yeah, we can go ahead and use that one now so the StatefulProcessorNode doesn't add the store twice (even if it's not problematic to do so)

Opened a quick hotfix PR for this: https://github.com/apache/kafka/pull/18015